dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/controller/ProcessDefinitionController.java

```diff
@@ -879,4 +880,29 @@ public Result viewVariables(@Parameter(hidden = true) @RequestAttribute(value =
         return returnDataList(result);
     }
 
+    /**
+     * Batch release process definition by code states
+     *
+     * @param loginUser   Login user
+     * @param projectCode Project code
+     * @param codeStates  Process definition code states
+     * @return Result of the batch release operation
+     */
+    @Operation(summary = "batchReleaseProcessDefinitionByCodeStates", description = "BATCH_RELEASE_PROCESS_DEFINITION_BY_CODE_STATES_NOTES")
+    @Parameters({
+            @Parameter(name = "codeStates", description = "Process definition code states", required = true, schema = @Schema(implementation = String.class))
+    })
+    @PostMapping(value = "/batch-release")
+    @ResponseStatus(HttpStatus.OK)
+    @ApiException(BATCH_RELEASE_PROCESS_DEFINE_BY_CODE_STATES_ERROR)
+    @AccessLogAnnotation(ignoreRequestArgs = "loginUser")
+    public Result batchReleaseProcessDefinitionByCodeStates(
+                                                            @Parameter(hidden = true) @RequestAttribute(value = Constants.SESSION_USER) User loginUser,
+                                                            @Parameter(name = "projectCode", description = "Project code", required = true) @PathVariable long projectCode,
+                                                            @RequestParam("codeStates") String codeStates) {
```

Comment:
perhaps we can split ```codeStates``` into two parameters, one is the code defined by all processes, and the other is the status of online or offline WDYT? cc @Radeity

Reply:
Using json `codeStates` is not a good way. A good way is as mentioned here
 https://github.com/apache/dolphinscheduler/pull/14089/files#r1193518157

Reply:
Okay, thank you. I made some changes over the weekend. I didn't have much time to come a few days ago. Just now, I saw a new submission, so I clicked on the update branch. hh

Reply:
> Okay, thank you. I made some changes over the weekend. I didn't have much time to come a few days ago. Just now, I saw a new submission, so I clicked on the update branch. hh

Open source projects do not have 996, So please take your time. ^_^